fanficfare/adapters/adapter_silmarillionwritersguildorg.py

```diff
@@ -0,0 +1,236 @@
+                           # sanitize_html() happier when description
+                           # is empty.
+        self.setDescription(url,summary)
+        
+        # no convenient way to extract metadata so bodge it by finding relevant identifier string and using next element as the data source
+
+        #get characters
+        charList = workDetails.findAll('a', href=re.compile(r'browse.php?type=characters'+"&charid=\d+$"))
+        charText = [char.string for char in charList]
+        self.chararacters = ', '.join(charText)
+        for char in charText:
+            self.story.addToList('characters',char.string)
+            
+        #get warnings
+        warnList = workDetails.findAll('a', href=re.compile(r'browse.php?type=characters'+"&charid=\d+$"))
+        warnText = [warn.string for warn in warnList]
+        self.warnings = ', '.join(charText)
+        for warn in charText:
+            self.story.addToList('warnings',char.string)
+            
+        #get genres
+        genresList = workDetails.findAll('a', href=re.compile(r'browse.php?type=class&type_id=1'+"&classid=\d+$"))
+        genrestext = [genre.string for genre in genresList]
+        self.genre = ', '.join(genrestext)
+        for genre in genrestext:
+            self.story.addToList('genre',genre.string)            
+        
+        #get rating
+        rating = workDetails.find('strong',text='Rated:').next_sibling.string
+        
+        #get completion status and correct for consistency with other adapters
+        if (workDetails.find('strong',text='Completed:').next_sibling.string).lower() == "yes":
+            status="Completed"
+        else:
+            status="In-Progress"
+        
+        #get wordcount
+        wordCount = workDetails.find('strong',text='Word count:').next_sibling.string
+        
+        #get published date, this works for some reason yet doesn't without the spaces in it
+        datePublished = workDetails.find('strong',text=' Published: ').next_sibling.string
+        
+        #get updated date
+        dateUpdated = workDetails.find('strong',text='Updated:').next_sibling.string
+        
+        # try setting metadata values on story, warn if it fails
+        try:
+            self.story.setMetadata('rating', rating)
+            self.story.setMetadata('status', status)
+            self.story.setMetadata('numWords', wordCount)
+            self.story.setMetadata('datePublished', makeDate(datePublished, self.dateformat))
+            self.story.setMetadata('dateUpdated', makeDate(dateUpdated, self.dateformat))
+        except Exception as e:
+            logger.warn("rating, status and/or datePublished parsing failed(%s) -- This can be caused by bad HTML in story description."%e)
+
+    # grab the text for an individual chapter.
+    def getChapterText(self, url):
+
+        logger.debug('Getting chapter text from: %s' % url)
+
+        if self.getConfig('is_adult'):
```

Comment:
Is this confirmAge POST needed for this site?  I haven't seen any chapters that require it, but I only checked a couple of 'Adult' stories.

Reply:
The site doesn't seem to have any adult checks on stories, so i've removed the POST code